### DISCUSSION TITLE
Using sub claim to lookup user

### DISCUSSION THREAD

**Discussion by tawaren** [JanssenProject/jans#8090] [Q&A] [answered]
Is their a way to get the user from the "sub" claim that can be found in the access & id tokens?

It seems that the "sub" claim is always generated on the fly and not stored and thus my preliminary conclusion is that I can not get a User from the "sub" value. However, I wanted to ask if this conclusion is correct or if I missed something?

**Comment by yuriyz** [comment, marked as answer]
AS supports both, persistent and algorithmic identifiers. If you set `client.subjectType=pairwise` and AS global configuration to `pairwiseIdType=persistent` then you can reference in identifer both client and user.

Please check these two code points
- https://github.com/JanssenProject/jans/blob/6c1caa1c4c92d28571f8589cd701e6885d4d85ef/jans-auth-server/server/src/main/java/io/jans/as/server/service/PairwiseIdentifierService.java#L68
- https://github.com/JanssenProject/jans/blob/546b1c23becbf9d00cdb4aaaeb086a44a9fb518b/jans-auth-server/persistence-model/src/main/java/io/jans/as/persistence/model/PairwiseIdentifier.java#L32

**Reply by tawaren** [reply]
This works, I know can resolve sub -> inum over a query using the EntryPersistenceManager.

**Comment by nynymike** [comment]
Also see the Docs on [Subject Identifiers](https://docs.jans.io/v1.1.0/admin/auth-server/openid-features/subject-identifiers/) 

Did you know you can search the docs? 

![image](https://github.com/JanssenProject/jans/assets/3717101/eab8091d-a1d8-4888-be72-e0de4b9c3206)
